apps/vmq_commons/src/gen_mqtt_client.erl

```diff
@@ -291,8 +294,16 @@ connected({unsubscribe, Topics} = Msg, State=#state{transport={Transport, _}, so
                                   State#state{msgid='++'(MsgId),
                                               info_fun=NewInfoFun})};
 
+connected({publish, PubReq}, #state{o_queue=#queue{size=Size} = _Q} = State) when Size > 0 ->
```

Comment:
As long as the queue (buffer) is > 0, we keep buffering. When the queue is empty, we switch to direct sending. Note that this is in connected state. This will not switch back to buffering when we are connected but overloaded (then we just load mailbox of gen_mqtt). This buffer just protects disconnected bridges.
We could probably change this & just send everything through the buffer though.